files/image_config/config-setup/config-setup

```diff
@@ -0,0 +1,399 @@
+    fi
+}
+
+# Check if updategraph service is administratively enabled
+updategraph_is_enabled()
+{
+    rv=1
+    if [ -e ${UPDATEGRAPH_CONF} ]; then
+        updategraph_mode=$(grep enabled ${UPDATEGRAPH_CONF} | head -n 1 | cut -f2 -d=)
+        [ "${updategraph_mode}" = "true" ] && rv = 0
+    fi
+    return $rv
+}
+
+# Disable updategraph admininistratively
+disable_updategraph()
+{
+    sed -i "/enabled=/d" ${UPDATEGRAPH_CONF}
+    echo "enabled=false" >> ${UPDATEGRAPH_CONF}
+}
+
+# Check if Zero Touch Provisioning is available and is administratively enabled
+ztp_is_enabled()
+{
+    rv=1
+    if [ -e /usr/bin/ztp ]; then
+        status=$(ztp status -c)
+        [ "$status" != "0:DISABLED" ] && [ "$status" != "" ] && rv=0
+    fi
+    return $rv
+}
+
+# Load requested SONiC configuration into config DB and initialize it
+# Usage: load_config <config_file>
+#
+#
+load_config()
+{
+    CONFIG_FILE=${1}
+    if [ "${CONFIG_FILE}" = "" ]; then
+        return 1
+    fi
+
+    redis-cli -n $CONFIG_DB_INDEX FLUSHDB
+    sonic-cfggen -j ${CONFIG_FILE} --write-to-db
+    if [ $? -ne 0 ]; then
+        return $?
+    fi
+
+    if [[ -x /usr/bin/db_migrator.py ]]; then
+        # Migrate the DB to the latest schema version if needed
+        /usr/bin/db_migrator.py -o migrate
+    fi
+
+    redis-cli -n $CONFIG_DB_INDEX SET "CONFIG_DB_INITIALIZED" "1"
+    return 0
+}
+
+
+# Generate requested SONiC configuration and save it as destination file
+# Usage: generate_config < factory | ztp > <destination_file>
+#
+#        factory - Create factory default configuration
+#        ztp - Create Zero Touch Provisioning Configuration
+#              used for provisioning data discovery.
+#
+generate_config()
+{
+    # Collect all information needed to generate configuration
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    PRESET=(`head -n 1 /usr/share/sonic/device/$PLATFORM/default_sku`)
+    HW_KEY=${PRESET[0]}
+    DEFAULT_PRESET=${PRESET[1]}
+
+    # Parse arguments passed
+    CONFIG_TYPE=$1
+    DEST_FILE=$2
+
+    if [ "$1" = "ztp" ]; then
+        /usr/lib/ztp/ztp-profile.sh create ${DEST_FILE}
+    elif [ "$1" = "factory" ]; then
+        rv=1
+
+        # Execute config initialization hooks
+        run_hookdir ${FACTORY_DEFAULT_HOOKS} ${CONFIG_SETUP_INITIALIZATION_FLAG}
+
+        # Use preset defined in default_sku
+        if [ ! -e ${DEST_FILE} ]; then
+            sonic-cfggen -H -k ${HW_KEY} --preset ${DEFAULT_PRESET} > ${DEST_FILE}
+            rv=$?
+            if [ $rv -ne 0 ]; then
+                return $rv
+            fi
+        fi
+    fi
+    return 0
+}
+
+# Create SONiC configuration for first time bootup
+#  - If ZTP is enabled, ZTP configuraion is created
+#  - If ZTP is disabled and updategraph is disabled, factory default configuration
+#    is created
+#  - If updategraph is enabled and ZTP is disabled, updategraph initializes
+#    configuration
+do_config_intialization()
```

Comment:
intialization looks like a typo

Reply:
Thanks for pointing it out. I created below PR to fix this issue.

https://github.com/Azure/sonic-buildimage/pull/4388